lib/internal/test_runner/runner.js

```diff
@@ -110,14 +110,17 @@ function filterExecArgv(arg, i, arr) {
   !ArrayPrototypeSome(kFilterArgValues, (p) => arg === p || (i > 0 && arr[i - 1] === p) || StringPrototypeStartsWith(arg, `${p}=`));
 }
 
-function getRunArgs({ path, inspectPort, testNamePatterns }) {
+function getRunArgs(path, { inspectPort, testNamePatterns, testOnly }) {
   const argv = ArrayPrototypeFilter(process.execArgv, filterExecArgv);
   if (isUsingInspector()) {
     ArrayPrototypePush(argv, `--inspect-port=${getInspectPort(inspectPort)}`);
   }
-  if (testNamePatterns) {
+  if (testNamePatterns != null) {
     ArrayPrototypeForEach(testNamePatterns, (pattern) => ArrayPrototypePush(argv, `--test-name-pattern=${pattern}`));
   }
+  if (testOnly != null) {
+    ArrayPrototypePush(argv, '--test-only');
+  }
```

Comment:
if it false you should not push it

Reply:
no:
```shell
./node --test-only=false --expose-internals -p "require('internal/options').getOptionValue('--test-only')"
```
prints
```js
true
```

Reply:
it can't come from the options in the `run` function?